Making it so we can target detail page vs show configuration

codyrancher opened the pull request:
<!-- This template is for Devs to give QA details before moving the issue To-Test -->
### Summary
Fixes #16179
Fixes #16189
<!-- Define findings related to the feature or bug issue. -->

### Occurred changes and/or fixed issues
Made is so we can now differentiate between detail pages and show configuration when adding tabs via extensions.

### Technical notes summary
This is done by using provide/inject to indicate to all ancestors that they're inside a resourcedetailpage. It's not the most elegant but it seems preferable over prop drilling especially when we don't control all detail pages that are rendered.

### Areas or cases that should be tested
Extension tabs  on the resource detail page, show configuration page, create and edit pages.

### Areas which could experience regressions
See above

### Screenshot/Video

https://github.com/user-attachments/assets/4ce9b812-d055-4ef0-861e-6d12585d71c1



### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone <!-- The milestone should automatically be assigned if the linked issue has one, but does not always happen (incorrectly linked, issue has no milestone, etc) -->
- [x] The PR template has been filled out
- [x] The PR has been self reviewed <!-- There are no TODOs, no incorrect files in the PR, all the required files are there, no commented out code, etc-->
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes
- [x] The PR has been reviewed in terms of Accessibility
- [x] The PR has considered, and if applicable tested with, the three Global Roles `Admin`, `Standard User` and `User Base`

richard-cox commented on shell/components/Tabbed/index.vue:
```diff
@@ -105,7 +106,8 @@ export default {
 
   data() {
-    const extensionTabs = this.showExtensionTabs ? getApplicableExtensionEnhancements(this, ExtensionPoint.TAB, TabLocation.RESOURCE_DETAIL, this.$route, this, this.extensionParams) || [] : [];
+    const location = this.isResourceDetailDrawer ? TabLocation.RESOURCE_SHOW_CONFIGURATION : TabLocation.RESOURCE_DETAIL;
```
@aalves08 @codyrancher just want to confirm my comment in https://github.com/rancher/dashboard/issues/16189#issuecomment-3661177094.

The issue needs updating to reflect that we need it primarily on the edit page and then the show config tab, so two tab locations. 

We probably need to be more explicit as well (if a --> y, if b --> x) instead of falling back on a location (if then else). Tabs can be used in a number of different places

aalves08 commented on shell/core/types.ts:
```diff
@@ -234,11 +235,6 @@ export interface ProductOptions {
-  /**
-   * Hide the product if the type is present (opposite of ifHaveType)
-   */
-  ifNotHaveType?: string | RegExp;
```
careful about this... probably is because it's missing a rebase or something. This was added quite recently. Should not be deleted

codyrancher commented:
> @codyrancher overall path looks great. I've tried the PR and it works just fine. Thank you very much! If you want I can take care of the documentation part in a separate PR 🙇

@aalves08  I cleaned it up and added some docs. If we do want to make separate location I'm happy to do that as well but it's up to you.

richard-cox reviewed (changes requested):
Re-iterating https://github.com/rancher/dashboard/pull/16203#discussion_r2627256347.

The developer experience is not good, we should have tab locations for each of the three places.

aalves08 commented:
> Re-iterating [#16203 (comment)](https://github.com/rancher/dashboard/pull/16203#discussion_r2627256347).
> 
> The developer experience is not good, we should have tab locations for each of the three places.

Created new isssue https://github.com/rancher/dashboard/issues/16218 to be tackled separately @richard-cox

aalves08 reviewed (changes requested):
minor changes, but let's try to get this in today 🙏  Thanks @codyrancher !

aalves08 reviewed (approved):
LGTM
